genesyscloud/resource_genesyscloud_credential.go

```diff
@@ -0,0 +1,189 @@
+package genesyscloud
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/platformclientv2"
+)
+
+func getAllCredentials(ctx context.Context, clientConfig *platformclientv2.Configuration) (ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(ResourceIDMetaMap)
+	integrationAPI := platformclientv2.NewIntegrationsApiWithConfig(clientConfig)
+
+	for pageNum := 1; ; pageNum++ {
+		credentials, _, err := integrationAPI.GetIntegrationsCredentials(pageNum, 100)
+		if err != nil {
+			return nil, diag.Errorf("Failed to get page of credentials: %v", err)
+		}
+
+		if credentials.Entities == nil || len(*credentials.Entities) == 0 {
+			break
+		}
+
+		for _, cred := range *credentials.Entities {
+			if cred.Name != nil { // Credential is possible to have no name
+				resources[*cred.Id] = &ResourceMeta{Name: *cred.Name}
+			}
+		}
+	}
+
+	return resources, nil
+}
+
+func credentialExporter() *ResourceExporter {
+	return &ResourceExporter{
+		GetResourcesFunc: getAllWithPooledClient(getAllCredentials),
+		RefAttrs:         map[string]*RefAttrSettings{}, // No Reference
+	}
+}
+
+func resourceCredential() *schema.Resource {
+	return &schema.Resource{
+		Description: "Genesys Cloud Credential",
+
+		CreateContext: createWithPooledClient(createCredential),
+		ReadContext:   readWithPooledClient(readCredential),
+		UpdateContext: updateWithPooledClient(updateCredential),
+		DeleteContext: deleteWithPooledClient(deleteCredential),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Credential name.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"credential_type_name": {
+				Description: "Credential type name.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"fields": {
+				Description: "Credential fields. Different credential types require different fields. Missing any correct required fields will result API request failure. Check out the specific credential type schema to find out what fields are required. ",
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Computed:    true,
```

Comment:
Since this can contain passwords, we should also add `Sensitive: true` to prevent it from displaying in logs.